src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -208,39 +208,65 @@ public void OnRenderCompleted(long incomingBatchId, string errorMessageOrNull)
                 return;
             }
 
+            // When clients send acks we know for sure they received and applied the batch.
+            // We send batches right away, and hold them in memory until we receive an ACK.
+            // If one or more client ACKs get lost (e.g., with long polling, client->server delivery is not guaranteed) we might receive an ack for a higher batch.
```

Comment:
Nit: long line (though please only fix if you're fixing other stuff here too)

Reply:
done